service/src/main/java/com/codeforcommunity/requester/S3Requester.java

```diff
@@ -178,21 +187,89 @@ private static String validateBase64ImageAndUploadToS3(
    */
   public static String validateUploadImageToS3LucyEvents(String eventTitle, String base64Encoding)
       throws BadRequestImageException, S3FailedUploadException {
-    String fileName = getImageFileNameWithoutExtension(eventTitle);
+    String fileName = getFileNameWithoutExtension(eventTitle, "_thumbnail");
     return validateBase64ImageAndUploadToS3(fileName, externs.getDirPublic(), base64Encoding);
   }
 
   /**
-   * Removes special characters, replaces spaces, and appends "_thumbnail".
+   * Removes special characters, replaces spaces, and appends a suffix.
    *
    * @param eventTitle the title of the event.
+   * @param suffix the suffix to be appended
    * @return the String for the image file name (without the file extension).
    */
-  public static String getImageFileNameWithoutExtension(String eventTitle) {
+  public static String getFileNameWithoutExtension(String eventTitle, String suffix) {
     String title =
         eventTitle.replaceAll(
             "[!@#$%^&*()=+./\\\\|<>`~\\[\\]{}?]", ""); // Remove special characters
-    return title.replace(" ", "_").toLowerCase()
-        + "_thumbnail"; // The desired name of the file in S3
+    return title.replace(" ", "_").toLowerCase() + suffix; // The desired name of the file in S3
+  }
+
+  /**
+   * Validate the given string encoding of HTML and upload it to the user upload S3 bucket.
```

Comment:
⛏️ Can we add a sentence here that says that a template will be overwritten if another template with the same name is uploaded?

Reply:
Done!